qiskit_addon_sqd/fermion.py

```diff
@@ -75,6 +75,7 @@ def solve_fermion(
     *,
     open_shell: bool = False,
     spin_sq: float | None = None,
+    shift: float | None = None,
```

Comment:
As I mentioned in the comment, we should make the default a number. From https://pyscf.org/_modules/pyscf/fci/addons.html it's either 0.2 or 0.1, but it's unclear which one. To be sure, we would need to run some code and print it out or use the debugger.

Reply:
We have been leaving it default, which leads me to believe [0.2 is what's been used](https://pyscf.org/pyscf_api_docs/pyscf.fci.html#pyscf.fci.addons.fix_spin).

Reply:
We have been calling `fix_spin_` with the default value, so I think [0.1 is what has been the default for us](https://pyscf.org/pyscf_api_docs/pyscf.fci.html#pyscf.fci.addons.fix_spin_)

Reply:
I decided to make the type hint `float` and cast to `np.float64` inside. The linter didn't want me to use `np.float64()` in the annotation. I could do `np.number | float`, but I figured this is fine